### PR TITLE
endpoints to delete specific batch

### PR DESCRIPTION
**Pull request by hemanth06912**
#107

### CODE REVIEW

File: src/server/api/controllers/delete-specific-batch.controller.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const knex = require('../../config/db');
+
+const deleteSpecificBatchById = async (batchId) => {
+  try {
+    await knex('batches_table')
```
Comment (amjadiftikhar) [inline src/server/api/controllers/delete-specific-batch.controller.js:5]: Is it a right name of the table? Should not be just 'batches'?

Comment (hemanth06912) [reply src/server/api/controllers/delete-specific-batch.controller.js:5]: yes it should be batches

**Comment by ricardoaguiar** [comment]
I have fixed a couple errors on the swagger.json and the router-api files. 
I get the 401 status on swagger but could not get the 200 with a successful delete.  Is there something missing to get the successful delete status?

File: src/server/config/swagger.json [hunk 2]
```diff
@@ -21,6 +21,14 @@
...
         ],
         "parameters": [
+          {
+            "name": "authorization",
+            "in": "header",
+            "description": "Firebase token",
+            "required": true,
+            "type": "string",
+            "default": "Bearer ENTER_FIREBASE_TOKEN"
+          },
```
Comment (sowmya1408) [inline src/server/config/swagger.json:31]: good use of firebase token object in swagger 👍

File: src/server/api/controllers/batch.controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const knex = require('../../config/db');
+const Roles = require('../../constants');
+
+const Error = require('../lib/utils/http-error');
+
+const deleteSpecificBatchById = async (batchId, res, next) => {
+  try {
+    const user = (await Roles.SUPER_ADMIN) === 'super_admin';
+    if (!user) return next(new Error('User have no rights to delete'));
```
Comment (sowmya1408) [inline src/server/api/controllers/batch.controller.js:9]: this code is not required, as it is taken care by the middleware checkIfAuthorized and I see that it is added to the endpoint.
Also please check there are two different files for same endpoint in controllers folder and also routes folder.

File: src/server/api/routes/delete-specfic-batch.router.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  checkIfAuthorized,
+} = require('../lib/middleware/authorization.middleware');
+
+const router = express.Router({ mergeParams: true });
+
+// controllers
+const deleteSpecificDataController = require('../controllers/delete-specific-batch.controller');
+
+// ENDPOINT: /api/example/ :DELETE
+router.delete('/:batchid', checkIfAuthorized(Roles.SUPER_ADMIN), (req, res) => {
```
Comment (sowmya1408) [inline src/server/api/routes/delete-specfic-batch.router.js:14]: good use of the middleware  :)

File: src/server/api/controllers/batch.controller.js (hunk 3 above)
Comment (hemanth06912) [reply src/server/api/controllers/batch.controller.js]: yes now it is resolved

**Comment by dpfernandes** [comment]
Approved

**Pull request merged**